subprojects/core/src/main/java/org/gradle/configuration/DefaultScriptPluginFactory.java

```diff
@@ -196,7 +196,7 @@ public void apply(final Object target) {
             String id = INTERNER.intern("cp_" + initialPassScriptTarget.getId());
             CompileOperation<?> initialOperation = new NoDataCompileOperation(id, CLASSPATH_COMPILE_STAGE, initialTransformer);
 
-            ScriptRunner<? extends BasicScript, ?> initialRunner = compiler.compile(scriptType, initialOperation, baseScope.getExportClassLoader(), Actions.doNothing());
+            ScriptRunner<? extends BasicScript, ?> initialRunner = compiler.compile(scriptType, initialOperation, targetScope, baseScope.getExportClassLoader(), Actions.doNothing());
```

Comment:
I think this should be `baseScope`, as the classpath portion of the script is compiled and executed against the Gradle API (the baseScope) rather than the stuff visible to the project (the targetScope).

Reply:
Done